editor/plugins/tiles/tile_data_editors.cpp

```diff
@@ -443,8 +446,15 @@ void GenericTilePolygonEditor::_snap_to_tile_shape(Point2 &r_point, float &r_cur
 	r_point = snapped_point;
 }
 
-void GenericTilePolygonEditor::_snap_to_half_pixel(Point2 &r_point) {
-	r_point = (r_point * 2).round() / 2.0;
+void GenericTilePolygonEditor::_snap_to_grid(Point2 &r_point) {
+	float snap_x = tile_set->get_tile_size().x / grid_snap->get_value();
+	float snap_y = tile_set->get_tile_size().y / grid_snap->get_value();
+	r_point = r_point.snapped(Vector2(snap_x, snap_y));
+
+	// The snap will be offset if grid_snap is odd. This fixes that, but it's weird.
+	if (int(grid_snap->get_value()) % 2 != 0) {
+		r_point -= Vector2(snap_x / 2, snap_y / 2);
+	}
```

Comment:
Wondered a moment on why this was needed. In fact it is the right thing to do as R_point is the position from the center of the texture (which is strange) and we want to align it with the borders and therefore when there is a odd number of subdivisions we need to offset it to get the right alignement.